core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/random.c

```diff
@@ -49,19 +49,19 @@ random_buf(void *buf, size_t len)
 
 #elif defined(BH_PLATFORM_WINDOWS)
 
-#include <wincrypt.h>
+#include <bcrypt.h>
+#pragma comment(lib, "Bcrypt.lib")
 
 void
 random_buf(void *buf, size_t len)
 {
-    static int crypt_initialized = 0;
-    static HCRYPTPROV provider;
-    if (!crypt_initialized) {
-        CryptAcquireContext(&provider, NULL, NULL, PROV_RSA_FULL,
-                            CRYPT_VERIFYCONTEXT);
-        crypt_initialized = 1;
+    NTSTATUS ret =
+        BCryptGenRandom(NULL, buf, (ULONG)len, BCRYPT_USE_SYSTEM_PREFERRED_RNG);
```

Comment:
according to the [doc](https://learn.microsoft.com/en-us/windows/win32/api/bcrypt/nf-bcrypt-bcryptgenrandom#return-value) the function fails on invalid algorithm's handle or invalid parameter. The former shouldn't be an issue because it's `NULL` in our case, but I the latter is vaguely defined, and I'm not 100% what it means (e.g. is there any min/max length of the buffer?). For that reason, I'd suggest changing the signature of the `random_buf` to return an error and instead of crashing through `abort()` return an error code (that can be then propagated to WASM in the `wasmtime_ssp_random_get()`.

Reply:
good idea, was thinking the same thing since crashing might be overkill if the function fails

Reply:
changed to return an error code